docs/advanced-scenarios.md

```diff
@@ -152,4 +152,177 @@ Google's `id_token` contains basic identity data like name and user ID, which al
+      interpolations
+    );
+    return result ? result : key;
+  }
+
+  /**
+   * @private
+   * Finds the translated value based on given key.
+   * @param {string} key key to be translated
+   * @param {*} obj translation table
+   */
+  findTranslation(key, obj, interpolations) {
+    let splitted = key.split('.');
+    for (let i = 0; i < splitted.length; i++) {
+      let key = splitted[i];
+      if (obj.hasOwnProperty(key) && typeof obj[key] === 'object') {
+        obj = obj[key];
+      } else {
+        if (interpolations)
+          return this.findInterpolations(obj[key], interpolations);
+        return obj[key];
+      }
+    }
+  }
+
+  /**
+   * @private
+   * Replaces values that are defiend in translation strings
+   * @param {string} key key to be translated
+   * @param {*} interpolations translation table
+   * @example
+   * findInterpolations('Environment {num}', {num: 1})
+   */
+  findInterpolations(value, interpolations) {
+    Object.keys(interpolations).forEach(item => {
+      value = value.replace(
+        new RegExp('{' + EscapingHelpers.escapeKeyForRegexp(item) + '}', 'gi'),
+        interpolations[item]
+      );
+    });
+    return value;
+  }
+}
+
+export const i18nProvider = new I18nProvider();
+```
+
+#### 2. Configure `customTranslationImplementation`
+
+Specify `settings.customTranslationImplementation`:
+
+```javascript
+import { i18nProvider } from './i18n-provider';
+const coreConfig = {
+  // ... partial setting ...
+  settings: {
+    customTranslationImplementation = i18nProvider
+  }
+}
+```
+
+#### 3. Load translations before `Luigi.setConfig()`
+
+Since translations may come from external sources, loaded asynchronously, we should load them ahead, before `Luigi.setConfig` is triggered. Be aware that other Luigi Core API functionality is only available after initialization.
+
+```javascript
+import { i18nProvider } from './i18n-provider';
+const coreConfig = { /* the whole configuration */ }
+
+i18nProvider.init().then(() => {
+  Luigi.setConfig(coreConfig);
+})
+```
+
+#### 4. Add locale change listener after Luigi initialization
+
+Specify `customTranslationImplementation`:
+
+```javascript
+import { i18nProvider } from './i18n-provider';
+const coreConfig = {
+  // ... partial setting ...
+  lifecycleHooks: {
+    luigiAfterInit: () => {
+      i18nProvider.afterInit();
+    }
+  }
+}
+```
+
+#### 5. Use translation as a node label
+
+You can use the translation keys as `node.label`, on other labels in the configuration or programmatically.
+
+```javascript
+const coreConfig = {
+  navigation: {
+    nodes: [{
+      label: 'MF1.LABEL1',
+      pathSegment: 'overview',
+      // ...
+    }],
+    contextSwitcher: {
+      defaultLabel: 'COMMON.SELECT_ENVIRONMENT',
+      options: () => [1,2,3,4,5].map(num => ({
+        label: Luigi.i18n().getTranslation('ENVIRONMENT_NUM', {num}),
+        pathValue: 'env' + num
+      }))
+    },
+  }
+}
+```
+
+#### Result
+
 <!-- accordion:end -->
```

Comment:
```suggestion
<!-- accordion:end -->
```

Are we still using accordions in this document? We can remove them if they affect search.

Reply:
I have implemented code which opens the accordions on search engine bots